include/zephyr/drivers/usb/udc.h

```diff
@@ -169,6 +169,8 @@ struct udc_event {
 	};
 	/** Pointer to device struct */
 	const struct device *dev;
+	/** Opaque pointer to higher layer context */
+	const void *ctx;
```

Comment:
Is this necessary here? Why not grab the pointer from `dev->data->event_ctx`?

Reply:
But then it needs to be abstracted to something like udc_get_event_context(). If it is part of the event, it is easier to handle and we are not introducing a new interface.

Reply:
Does it need to be abstracted? Can't the `event_cb` simply take the `event_ctx` as extra parameter?

Reply:
No, it can only put a message into a queue.  I added udc_get_event_ctx(), seems to be more performant.